nautobot/dcim/models/device_components.py

```diff
@@ -376,6 +376,9 @@ def get_power_draw(self):
             "maximum": self.maximum_draw or 0,
             "outlet_count": PowerOutlet.objects.filter(power_port=self).count(),
             "legs": [],
+            "utilization_data": UtilizationData(
+                numerator=self.allocated_draw or 0, denominator=self.connected_endpoint.available_power or 0
```

Comment:
Sorry to keep revising my own suggestion, but are we guaranteed to *have* a non-null `self.connected_endpoint` here? Might be safer to check it to be sure.